common/lib/capa/capa/tests/test_inputtypes.py

```diff
@@ -747,29 +748,30 @@ def test_matlab_api_key(self):
+        self.assertIn('<div>{', output_string)
+        output_string = output_string.replace('}</div>', '')
+        output_string = output_string.replace('<div>{', '')
+        output_list = output_string.split(',')
+        for index, value in enumerate(output_list):
+            output_list[index] = value.strip()
+
+        expected_string = u"""
+        \'matlab_editor_js\': \'/dummy-static/js/vendor/CodeMirror/octave.js\',
+        \'value\': \'print "good evening"\', \'hidden\': \'\',
+        \'msg\': u\'Submitted. As soon as a response is returned, this message will be replaced by that feedback.\',
+        \'status\': Status(\'queued\'), \'response_data\': {}, \'queue_msg\': \'\', \'mode\': \'\',
+        \'id\': \'prob_1_2\', \'queue_len\': \'3\', \'tabsize\': 4, \'STATIC_URL\': \'/dummy-static/\',
+        \'linenumbers\': \'true\', \'cols\': \'80\', \'button_enabled\': True, \'rows\': \'10\',
+        \'describedby_html\': Markup(u\'aria-describedby="status_prob_1_2"\')"""
+        expected_list = (textwrap.dedent(expected_string).replace('\n', ' ').strip()).split(',')
+        for index, value in enumerate(expected_list):
+            expected_list[index] = value.strip()
         if six.PY2:
-            expected_string = """
-            <div>{\'status\': Status(\'queued\'), \'button_enabled\': True, \'rows\': \'10\',
-            \'queue_len\': \'3\', \'mode\': \'\', \'tabsize\': 4, \'cols\': \'80\', \'STATIC_URL\': \'/dummy-static/\',
-            \'linenumbers\': \'true\', \'queue_msg\': \'\', \'value\': \'print "good evening"\',
-            \'msg\': u\'Submitted. As soon as a response is returned, this message will be replaced by that feedback.\',
-            \'matlab_editor_js\': \'/dummy-static/js/vendor/CodeMirror/octave.js\',
-            \'hidden\': \'\', \'id\': \'prob_1_2\',
-            \'describedby_html\': Markup(u\'aria-describedby="status_prob_1_2"\'), \'response_data\': {}}</div>
-            """
+            self.assertItemsEqual(output_list, expected_list)
         else:
-            expected_string = """
-            <div>{\'matlab_editor_js\': \'/dummy-static/js/vendor/CodeMirror/octave.js\',
-            \'value\': \'print "good evening"\', \'hidden\': \'\',
-            \'msg\': \'Submitted. As soon as a response is returned, this message will be replaced by that feedback.\',
-            \'status\': Status(\'queued\'), \'response_data\': {}, \'queue_msg\': \'\', \'mode\': \'\',
-            \'id\': \'prob_1_2\', \'queue_len\': \'3\', \'tabsize\': 4, \'STATIC_URL\': \'/dummy-static/\',
-            \'linenumbers\': \'true\', \'cols\': \'80\', \'button_enabled\': True, \'rows\': \'10\',
-            \'describedby_html\': Markup(\'aria-describedby="status_prob_1_2"\')}</div>"""
-        self.assertEqual(
-            etree.tostring(output).decode('utf-8'),
-            textwrap.dedent(expected_string).replace('\n', ' ').strip(),
-        )
+            self.assertCountEqual(output_list, expected_list)
```

Comment:
You could use [six.assertCountEqual](https://six.readthedocs.io/#six.assertCountEqual) instead of an explicit condition here.